### PR TITLE
[Localization] #1761 Korean trainer dialogue (villain)

### PR DESCRIPTION
**Pull request by sodaMelon**
[Localization] #1761 Korean trainer dialogue (villain)

## What are the changes?

Translating locales\ko\dialogue.ts into Korean.
This PR is added korean dialogue for villain

@sodaMelon @Ohmry @returntoice now participating in https://github.com/pagefaultgames/pokerogue/issues/1761 translating project.

## Why am I doing these changes?

Few trainers's dialogue have been translated into Korean yet.

## What did change?

I translated **the  villain trainer battles' dialogues.**
(It hasn't been released yet in game)

빌런 (로켓단, 마그마단, 갤럭시단..........들)의 스크립트를 번역했습니다.
아직 게임에서 등장하지는 않았지만 아침에 ice님 PR 보고 해도 되겠다 싶어서 번역했어요~. 

천천히 보시고, 부자연스럽다면 편하게 Suggestion 주세요!

### Screenshots/Videos

( very hard to meet them.... and it's dosen't matter)

## How to test the changes?

It seems there won't be much to test since it's just a translation of an existing .ts file. You can check it when you meet the respective trainers.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~~[ ] Are the changes visual?~~
  - ~~[ ] Have I provided screenshots/videos of the changes?~~

### CODE REVIEW

File: src/locales/ko/dialogue.ts [hunk 1]
```diff
@@ -385,182 +385,182 @@ export const PGMdialogue: DialogueTranslationEntries = {
...
-      1: "I won't allow anyone to stop me! No matter who does what!"
+      1: "누구도 저를 막을 수 없습니다! 무슨 짓을 하더라도!"
     },
     "victory": {
-      1: "How can this be? I'm the creator of Team Plasma! I'm perfect!"
+      1: "…어떻게 된 거지? 이 몸은 플라스마단을 만들어낸 완벽한 남자라고! "
     },
     "defeat": {
-      1: "I am the perfect ruler of a perfect new world! Mwa ha ha!"
+      1: "세계를 변하게 할 완벽한 지배자란 말이다! 마 하 하!"
```
Comment (sodaMelon) [inline src/locales/ko/dialogue.ts:530]: 게치스 웃음소리가 기억이 안나서...😅 혹시 아시는 분 있으면 제안 부탁드립니다

File: src/locales/ko/dialogue.ts [hunk 2]
```diff
@@ -385,182 +385,182 @@ export const PGMdialogue: DialogueTranslationEntries = {
...
     },
     "defeat": {
-      1: "Kyurem! Use Absofusion!"
+      1: "큐레무! 합체해라!"
     }
   },
   "flare_boss_lysandre_1": {
     "encounter": {
-      1: "Do you want to stop me? Show me in battle."
+      1: "나를 막겠다고? 배틀로 증병하도록."
```
Comment (sodaMelon) [inline src/locales/ko/dialogue.ts:546]:
```suggestion
      1: "나를 막겠다고? 배틀로 증명하도록."
```
오타 수정 예정입니다

File: src/locales/ko/dialogue.ts (hunk 1 above)
Comment (returntoice) [inline src/locales/ko/dialogue.ts:530]:
```suggestion
      1: "세계를 바꿀 완전한 지배자가 아니었단 말인가!? …흐하하!"
```
https://youtu.be/jx649erM1k4?si=gu7bxoALb5RkEEaB&t=763
https://youtu.be/jx649erM1k4?si=JHr9zF_TYuM6CNPm&t=774
블화 원본 대사를 찾아왔습니다

**Review by returntoice — CHANGES_REQUESTED** [review]

**Review by returntoice — APPROVED** [review]

File: src/locales/ko/dialogue.ts (hunk 1 above)
Comment (sodaMelon) [reply src/locales/ko/dialogue.ts]: 오 감사합니다 ㅋㅋㅋ 마 하 하 웃는 게 로컬라이징이 어떻게 됐나...?
바하하하 (1세대 고스트 타입 사천왕, 국화) 후하하하 (1세대 독타입 관장/2세대 사천왕, 독수) 같은경우는 특이한 웃음소리를 잘 살려주더라구요 .. 국제적으로 200x년시절부터... 
그래서 결정도 못내리고 + 게치스가 웃는장면이 있었던가 고민하고 있었어요 ㅋㅋㅋㅋ 감사합니다~~!

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**